extensions/SharkPool/BetterInput.js

```diff
@@ -0,0 +1,1305 @@
+          overlay.style.opacity = "0";
+        }
+
+        if (this.enterEffect === "growIn" || this.enterEffect === "Both") {
+          overlay.style.transform = "scale(0)";
+          overlay.style.transition = "opacity 0.3s";
+        }
+
+        const step = () => {
+          if (
+            (this.enterEffect === "fadeIn" || this.enterEffect === "Both") &&
+            currentOpacity < 100
+          ) {
+            currentOpacity += this.enterSpeed;
+            overlay.style.opacity = `${currentOpacity}%`;
+          }
+
+          if (
+            (this.enterEffect === "growIn" || this.enterEffect === "Both") &&
+            currentScale < 100
+          ) {
+            currentScale += this.enterSpeed;
+            overlay.style.transform = `scale(${currentScale / 100})`;
+          }
+
+          if (
+            (this.enterEffect === "Both" &&
+              (currentOpacity < 100 || currentScale < 100)) ||
+            (this.enterEffect === "growIn" && currentScale < 100) ||
+            (this.enterEffect !== "Both" && currentOpacity < 100)
+          ) {
+            requestAnimationFrame(step);
+          }
+        };
+
+        if (this.enterEffect === "growIn" || this.enterEffect === "Both") {
+          overlay.style.transition = "opacity 0.3s";
+        }
+
+        requestAnimationFrame(step);
+      } else {
+        overlay.style.opacity = "1";
+      }
+    }
+
+    applyExitEffect(overlay) {
+      let currentScale = 100;
+      let currentOpacity = 0;
+
+      if (this.exitEffect !== "None") {
+        if (this.exitEffect === "fadeOut") {
+          overlay.style.transition = "opacity 0.5s";
+        } else {
+          overlay.style.transition = "transform 0.5s, opacity 0.5s";
+        }
+
+        const step = () => {
+          if (
+            (this.exitEffect === "fadeOut" || this.exitEffect === "Both") &&
+            currentOpacity < 100
+          ) {
+            currentOpacity -= this.exitSpeed;
+            overlay.style.opacity = `${currentOpacity}%`;
+          }
+
+          if (
+            (this.exitEffect === "shrinkOut" || this.exitEffect === "Both") &&
+            currentScale > 0
+          ) {
+            currentScale -= this.exitSpeed;
+            overlay.style.transform = `scale(${currentScale / 100})`;
+          }
+
+          if (
+            (this.exitEffect === "Both" &&
+              (currentOpacity > -200 || currentScale > -100)) ||
+            (this.exitEffect === "shrinkOut" && currentScale > -100) ||
+            (this.exitEffect !== "Both" && currentOpacity > -200)
+          ) {
+            requestAnimationFrame(step);
+          } else {
+            document.body.removeChild(overlay);
+          }
+        };
+
+        requestAnimationFrame(step);
+      } else {
+        document.body.removeChild(overlay);
+      }
+    }
+
+    askAndWait(args) {
+      if (this.askBoxCount < this.maxBoxCount) {
```

Comment:
the normal Scratch ask blocks will wait for the old prompt to disappear, then show the new one. would it make sense to do that here too?

Reply:
No need for it with the new block in this update. Since I want to allow people to have more than 1 ask box, I just added a <is waiting for input?> boolean instead. If users really wanted to wait for the old ask input to disappear before asking for the new one, they can make it themselves with the ask block. Its better this way since this extension is supposed to be a fully customizable expansion + remodel of the Ask block.